src/mon/FSCommands.cc

```diff
@@ -1526,6 +1514,13 @@ int FileSystemCommandHandler::_check_pool(
 
   const string& pool_name = osd_map.get_pool_name(pool_id);
 
+  if (!allow_overlay && pool->application_metadata.count(pg_pool_t::APPLICATION_NAME_CEPHFS) != 0) {
```

Comment:
`application_metadata.count()` is a bit expensive than `application_metadata.find()`

Reply:
Done